tensorizer/stream_io.py

```diff
@@ -897,23 +908,24 @@ def _ensure_https_endpoint(endpoint: str):
         raise ValueError("Non-HTTPS endpoint URLs are not allowed.")
 
 
+def _is_caios(endpoint: str) -> bool:
+    host = urlparse(endpoint if "//" in endpoint else "//" + endpoint).netloc
+    return host.lower() in {"cwobject.com", "cwlota.com"}
+
+
 def _new_s3_client(
-    s3_access_key_id: str,
-    s3_secret_access_key: str,
-    s3_endpoint: str,
+    s3_access_key_id: Optional[str],
+    s3_secret_access_key: Optional[str],
+    s3_endpoint: Optional[str],
     s3_region_name: Optional[str] = None,
     s3_signature_version: Optional[str] = None,
 ):
-    if s3_secret_access_key is None:
-        raise TypeError("No secret key provided")
-    if s3_access_key_id is None:
-        raise TypeError("No access key provided")
-    if s3_endpoint is None:
-        raise TypeError("No S3 endpoint provided")
-
     config_args = dict(user_agent=_BOTO_USER_AGENT)
     auth_args = {}
 
+    if _is_caios(s3_endpoint):
```

Comment:
This seems like something that should only be set if not already explicitly set by the configuration from boto3.

Reply:
It is something that can be set in an `.aws/config` file for Boto3, but since:

1. This is the only valid option for these endpoints anyway
2. (AFAIK) We don't provide users a generated `.aws/config` file or template to use in our web console nor in our CW documentation, or explain how to configure this in our CW documentation

It's good to ensure this here for these specific endpoints. The default behaviour in Boto3 is still used for other endpoints. Though we ought to update our documentation to clarify this as well.

Reply:
1. That is fair, if boto3 has similar custom configurations then I have less pushback on this.
2. We actually do have our CW documentation https://docs.coreweave.com/docs/products/storage/object-storage/how-to/manage-buckets#manage-buckets-with-aws-cli.

Reply:
Those docs leave a bit to be desired yes.